plugins/dell-dock/fu-plugin-dell-dock.c

```diff
@@ -226,7 +206,7 @@ fu_plugin_dell_dock_separate_activation(FuPlugin *plugin)
 		FuDevice *device_tmp = g_ptr_array_index(devices, i);
 		if (FU_IS_DELL_DOCK_EC(device_tmp))
 			device_ec = device_tmp;
-		else if (FU_IS_DELL_DOCK_USB4(device_tmp))
+		else if (g_strcmp0(fu_device_get_logical_id(device_tmp), "usb4") == 0)
```

Comment:
Something we need to think about now with this being a generic plugin is what happens if two different USB4 docks are plugged in simultaneously?  Do the right things get done?